### PR TITLE
Initial support for kernel 5.15 via git clone

### PR DESCRIPTION
**Pull request by snobleaws**
Add support for Linux kernels from git to kbuild.mk
Clone https://github.com/dentproject/linux 5.15.y dent specific branch into build
Build for NXP rdb
Tested on NXP ls1043ardb
root@localhost:~# cat /etc/onl/platform
arm64-nxp-ls1043ardb-r0
root@localhost:~# uname -a
Linux localhost 5.15.11-g498d93b3aea9-dirty #1 SMP PREEMPT Sun Jan 23 23:43:15 UTC 2022 aarch64 GNU/Linux
Signed-off-by: Steve Noble <snoble@sonn.com>

### CODE REVIEW

**Comment by sonoble** [comment]
I am interested in feedback on the implementation. It could use more error checking (did you define the branch) or just default to the main branch. I may also be missing some nuances of makefiles.

**Review by eladnachman — APPROVED** [review]

File: setup.env [hunk 1]
```diff
@@ -7,7 +7,7 @@
 ############################################################
 
 # Fix expired letsencrypt root certificate https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/
-sudo apt-get update; sudo apt-get install -y libgnutls30
+sudo apt-get update; sudo apt-get install -y libgnutls30 pxz
```
Comment (taraschornyiplv) [inline setup.env:10]: I think it is better to add this to a Dockerfile and switch from ONL to dent project builder

File: packages/base/any/kernels/5.15-lts/kconfig.mk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+THIS_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+K_MAJOR_VERSION := 5
+K_PATCH_LEVEL := 15
+K_SUB_LEVEL := 11
```
Comment (taraschornyiplv) [inline packages/base/any/kernels/5.15-lts/kconfig.mk:4]: can we get these numbers from https://github.com/dentproject/linux/blob/dent-linux-5.15.y/Makefile#L4

File: setup.env (hunk 1 above)
Comment (snobleaws) [reply setup.env:10]: You are correct, this is a quick response to the security issue in gnutls, I was also experimenting with pxz to see if it helped speed up the build.  I will remove them both.  I need to submit the 1.7 Dockerfile which updates this.  I will remove this from the PR.

**Review by taraschornyiplv — APPROVED** [review]

File: packages/base/any/kernels/5.15-lts/kconfig.mk (hunk 2 above)
Comment (robimarko) [reply packages/base/any/kernels/5.15-lts/kconfig.mk:4]: You can parse it from the cloned kernel source instead of hardcoding and having to update it.

**Pull request merged**